python_modules/libraries/dagster-managed-elements/dagster_managed_elements/types.py

```diff
@@ -0,0 +1,237 @@
+import enum
+from abc import ABC, abstractmethod
+from typing import Any, List, NamedTuple, Optional, OrderedDict, Tuple, Union
+
+import click
+
+import dagster._check as check
+
+
+class ManagedElementError(enum.Enum):
+    CANNOT_CONNECT = "cannot_connect"
+
+
+SANITIZE_KEY_KEYWORDS = ["password", "token", "secret"]
+
+
+def _sanitize(key: str, value: str):
+    """
+    Rudamentary sanitization of values so we can avoid printing passwords
+    to the console.
+    """
+    if any(keyword in key for keyword in SANITIZE_KEY_KEYWORDS):
+        return "**********"
+    return value
+
+
+DiffData = Tuple[str, str]
+ModifiedDiffData = Tuple[str, str, str]
+
+
+class ManagedElementDiff(
+    NamedTuple(
+        "_ManagedElementDiff",
+        [
+            ("additions", List[DiffData]),
+            ("deletions", List[DiffData]),
+            ("modifications", List[ModifiedDiffData]),
+            ("nested", OrderedDict[str, "ManagedElementDiff"]),
+        ],
+    )
+):
+    """
+    Utility class representing the diff between configured and deployed managed element. Can be rendered to a
+    color-coded, user-friendly string.
+    """
+
+    def __new__(
+        cls,
+        additions: Optional[List[DiffData]] = None,
+        deletions: Optional[List[DiffData]] = None,
+        modifications: Optional[List[ModifiedDiffData]] = None,
+    ):
+        check.opt_list_param(additions, "additions", of_type=DiffData)
+        check.opt_list_param(deletions, "deletions", of_type=DiffData)
+        check.opt_list_param(modifications, "modifications", of_type=ModifiedDiffData)
+
+        return super().__new__(
+            cls,
+            additions or [],
```

Comment:
```suggestion
            check.opt_list_param(additions, "additions", of_type=DiffData),
```
`opt_list_param` does this for you.
https://github.com/dagster-io/dagster/blob/master/python_modules/dagster/dagster/_check/__init__.py#L747